# Исследование надежности заемщиков


Во второй части проекта вы выполните шаги 3 и 4. Их вручную проверит ревьюер.
Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом. 



## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [1]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [2]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [4]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [5]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [6]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [7]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть. Тем более этот столбец не понадобится вам для исследования.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [8]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [9]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [10]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [11]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [12]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [13]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру. Проверьте остальные столбцы.**

In [14]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [15]:
data.duplicated().sum()

71

In [16]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [17]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [18]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [19]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [20]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [21]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

In [22]:
# Считаем количество должников в каждой группе по количеству детей
debt_count_children = data.groupby('children')['debt'].sum()



In [23]:
# Определяем общее количество людей в каждой группе по количеству детей
total_count_children = data.groupby('children')['debt'].count()



In [24]:
# Вычисляем долю должников в каждой группе
debt_ratio_children = debt_count_children / total_count_children



In [25]:
# Выводим результат, отсортированный по убыванию доли должников
debt_ratio_children = debt_ratio_children.sort_values(ascending=False)
print(debt_ratio_children)

children
4    0.097561
2    0.094542
1    0.092346
3    0.081818
0    0.075438
5    0.000000
Name: debt, dtype: float64


**Вывод:** Результаты, которые мы получили, показывают долю должников для каждой категории `children` (количества детей). Можем сделать следующие выводы:

- **4 детей: 9.76%** — Доля должников среди людей с 4 детьми составляет 9.76%. Это самая высокая доля среди всех категорий.
- **2 детей: 9.45%** — Люди с 2 детьми также имеют относительно высокую долю должников, близкую к 9.45%.
- **1 ребенок: 9.23%** — Люди с 1 ребенком имеют чуть более низкую долю должников — 9.23%.
- **3 детей: 8.18%** — Доля должников среди людей с 3 детьми составляет 8.18%.
- **0 детей: 7.54%** — Самая низкая доля должников наблюдается среди людей без детей — 7.54%.
- **5 детей: 0%** — Нет должников среди людей с 5 детьми в предоставленных данных.

Как мы можем интерпретировать полученные результаты:

1. **Люди с 4 детьми** имеют самую высокую долю должников, что может свидетельствовать о большей финансовой нагрузке, связанной с большим количеством детей.
  
2. **Люди с 5 детьми** не имеют должников в данных, но это может быть связано с малым количеством наблюдений в этой группе (например, если таких людей очень мало).

3. **Отсутствие детей** связано с самой низкой долей должников, что может указывать на меньшие финансовые обязательства.

Выводы:

Эти данные показывают, что, хотя абсолютное количество должников может быть выше в группах с меньшим количеством детей, группы с 4 детьми потенциально более рискованны для банка с точки зрения вероятности возникновения долга. Информация о доле должников в каждой группе может помочь банку принимать более взвешенные решения при оценке кредитных рисков.


#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [26]:
# Считаем количество должников в каждой группе по семейному положению
debt_count_status = data.groupby('family_status')['debt'].sum()


In [27]:
# Определяем общее количество людей в каждой группе по по семейному положению
total_count_status = data.groupby('family_status')['debt'].count()


In [28]:
# Вычисляем долю должников в каждой группе
debt_ratio_status = debt_count_status / total_count_status


In [29]:
# Выводим результат, отсортированный по убыванию доли должников
debt_ratio_status = debt_ratio_status.sort_values(ascending=False)
print(debt_ratio_status)

family_status
Не женат / не замужем    0.097639
гражданский брак         0.093130
женат / замужем          0.075606
в разводе                0.070648
вдовец / вдова           0.066246
Name: debt, dtype: float64


**Вывод:** Полученные данные показывают, как семейное положение влияет на вероятность наличия долгов:

1. **Не женат / не замужем (9.76%)**: У одиноких людей больше шансов иметь долги. Это может быть связано с меньшей финансовой стабильностью.
  
2. **Гражданский брак (9.31%)**: Люди, живущие вместе без брака, тоже часто имеют долги, хотя и немного реже, чем одинокие.

3. **Женат / замужем (7.56%)**: Супруги реже имеют долги, вероятно, благодаря совместной финансовой поддержке и стабильности.

4. **В разводе (7.06%)**: Разведённые люди имеют ещё меньше долгов, возможно, они стали осторожнее в финансах.

5. **Вдовец / вдова (6.62%)**: Вдовы и вдовцы реже имеют долги, возможно из-за возраста и накопленного опыта.

Вероятность наличия долгов снижается по мере перехода от "не женат / не замужем" к "вдовец / вдова". Это может свидетельствовать о том, что стабильность и ответственность, связанные с браком, а также возраст и жизненный опыт, оказывают значительное влияние на финансовое состояние и способность управлять долгами.

Таким образом, семейное положение может быть важным фактором при оценке кредитных рисков. Люди, которые не состоят в браке или находятся в гражданском браке, могут представлять собой более высокий риск для кредиторов по сравнению с теми, кто состоит в зарегистрированном браке, разведены или овдовели.

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [30]:
# Считаем количество должников в каждой группе по уровню дохода
debt_count_income = data.groupby('total_income_category')['debt'].sum()
debt_count_income

total_income_category
A       2
B     354
C    1353
D      21
E       2
Name: debt, dtype: int64

In [31]:
# Определяем общее количество людей в каждой группе по уровню дохода
total_count_income = data.groupby('total_income_category')['debt'].count()
total_count_income


total_income_category
A       25
B     5014
C    15921
D      349
E       22
Name: debt, dtype: int64

In [32]:
# Вычисляем долю должников в каждой группе
debt_ratio_income = debt_count_income / total_count_income


In [33]:
# Выводим результат, отсортированный по убыванию доли должников
debt_ratio_income = debt_ratio_income.sort_values(ascending=False)
print(debt_ratio_income)

total_income_category
E    0.090909
C    0.084982
A    0.080000
B    0.070602
D    0.060172
Name: debt, dtype: float64


**Вывод:** 

*Анализ размеров групп и выводы:*

1. **Размеры групп**:
   - **Категория A** (доход 1,000,001 и выше): 25 человек, из них 2 должника.
   - **Категория B** (доход 200,001–1,000,000): 5014 человек, из них 354 должника.
   - **Категория C** (доход 50,001–200,000): 15,921 человек, из них 1353 должника.
   - **Категория D** (доход 30,001–50,000): 349 человек, из них 21 должник.
   - **Категория E** (доход до 30,000): 22 человека, из них 2 должника.
   

**Выводы:**

1. **Достаточность числа клиентов для построения выводов**:
   - **Категории B и C** имеют достаточно большие выборки (5014 и 15,921 человек соответственно) для построения надежных выводов о зависимости возврата кредита от уровня доходов.
   - **Категории D** (349 человек) также может быть достаточно для анализа, хотя выборка существенно меньше, чем у категорий B и C, что может повлиять на надежность выводов о зависимости.
   - **Категории A и E** (25 и 22 человека соответственно) слишком малы для статистически значимых выводов. Только 2 должника в каждой из этих категорий не позволяют делать уверенные заключения о реальной зависимости между уровнем дохода и возвратом кредита.

2. **Зависимость между уровнем дохода и возвратом кредита**:
   - **Категории B и D** (доход от 30,001 до 1,000,000) показывают более низкую долю должников (7.06% и 6.02% соответственно), что указывает на более ответственное управление кредитами в этих группах.
   - **Категория C** (доход 50,001–200,000) имеет более высокую долю должников — 8.50%, что может указывать на повышенные риски в этой группе.
   - **Категории A и E** из-за малой выборки не позволяют сделать надежные выводы. Несмотря на то что доля должников в категории E (9.09%) и A (8.00%) выше, чем в категориях B и D, эти данные не показательны из-за малого числа наблюдений.

Итог:
Зависимость между уровнем дохода и возвратом кредита в срок действительно существует, но она лучше видна в более крупных группах (B и C). Категории B и D, с доходами от 30,001 до 1,000,000, демонстрируют меньшую вероятность просрочки по кредитам. Для категорий A и E выводы нельзя считать надежными из-за малых размеров выборки.

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

In [34]:
# Считаем количество должников в каждой группе в зависимости от цели кредита
debt_count_purpose = data.groupby('purpose_category')['debt'].sum()


In [35]:
# Определяем общее количество людей в каждой группе в зависимости от цели кредита
total_count_purpose = data.groupby('purpose_category')['debt'].count()


In [36]:
# Вычисляем долю должников в каждой группе
debt_ratio_purpose = debt_count_purpose / total_count_purpose


In [37]:
# Выводим результат, отсортированный по убыванию доли должников
debt_ratio_purpose = debt_ratio_purpose.sort_values(ascending=False)
print(debt_ratio_purpose)

purpose_category
операции с автомобилем      0.093480
получение образования       0.092528
проведение свадьбы          0.079118
операции с недвижимостью    0.072551
Name: debt, dtype: float64


**Вывод:** Данные показывают, как цель кредита влияет на вероятность просрочки:


1. **операции с автомобилем (9.35%)**: Самая высокая вероятность просрочки, вероятно, из-за непредвиденных расходов.
2. **получение образования (9.25%)**: Часто просрочки, так как доход от образования приходит не сразу.
3. **проведение свадьбы (7.91%)**: Реже просрочки, так как расходы планируются заранее.
4. **операции с недвижимостью (7.26%)**: Самая низкая вероятность просрочки, это может быть связано с тем, что недвижимость является значимым и долговременным активом, а заемщики более осторожны и ответственны при его покупке.

**Итог**:
Цель кредита существенно влияет на вероятность его просрочки. Кредиты на автомобиль и образование чаще просрочиваются, вероятно, из-за высокой стоимости и финансовой нагрузки, тогда как кредиты на свадьбу и недвижимость реже приводят к проблемам с возвратом, поскольку эти расходы обычно лучше планируются и связаны с важными жизненными решениями.

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

*Ответ:* Пропуски в данных могут возникать по различным причинам. В нашем случае возможные причины появления пропусков в переменных `days_employed` и `total_income` могут быть следующими:

1. **Отсутствие трудового стажа**: Пропуски в `days_employed` могут возникать у людей, у которых нет официального трудового стажа. Это касается людей, которые никогда не работали официально, не имели стабильной работы или, возможно, являются студентами или пенсионерами.

2. **Расчет дохода на основании стажа**: Поскольку количество пропусков в `days_employed` и `total_income` идентичны, можно предположить, что доход рассчитывается на основании трудового стажа. Если стаж не указан, то доход также не может быть рассчитан, что приводит к пропускам в обеих переменных.

3. **Случайные пропуски или технические ошибки**: Пропуски могут быть связаны с техническими ошибками при сборе или выгрузке данных. Например, данные могли быть не полностью заполнены или не корректно перенесены из одной системы в другую.

4. **Ошибки ввода данных**: В некоторых случаях сотрудники, вводящие данные, могли случайно пропустить часть информации, особенно если она не является обязательной для заполнения.

5. **Отсутствие актуальной информации**: В некоторых случаях данные могут отсутствовать, потому что клиент не предоставил актуальную информацию о своём стаже или доходе на момент подачи заявки на кредит.

Эти причины должны быть учтены при дальнейшей работе с данными, чтобы минимизировать их влияние на результаты анализа.

#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

**Ответ:** Заполнение пропусков медианным значением является предпочтительным решением для количественных переменных по нескольким причинам:

Устойчивость к выбросам: Медианное значение устойчиво к аномально высоким или низким значениям (выбросам) в данных. В отличие от среднего значения, которое может сильно сместиться в сторону выбросов, медиана отражает центральное значение распределения, что делает её более надежной для замены пропусков.

Представление наиболее частых данных: Медиана разделяет данные на две равные части, таким образом представляя типичное значение, которое находится в середине распределения. Это особенно важно в ситуациях, когда данные имеют смещённое распределение или наличие выбросов, которые искажают среднее значение.

Сохранение реальности данных: Заполнение пропусков медианой позволяет сохранить реалистичное представление о данных, особенно в ситуациях, когда данные не распределены нормально. Например, если зарплаты большинства сотрудников находятся на уровне 100 000, а у немногих — значительно выше, медиана лучше отражает типичную зарплату в компании, чем среднее значение.

Снижение искажения данных: Заполнение пропусков медианой помогает избежать искажения данных, которое могло бы возникнуть при использовании среднего значения. В результате, медиана помогает сохранить реальную структуру данных и обеспечивает более точные результаты анализа.

Таким образом, заполнение пропусков медианным значением помогает избежать искажений, которые могут возникнуть при использовании среднего значения, и обеспечивает более точное и стабильное представление данных, особенно в присутствии выбросов или смещённого распределения.

### Шаг 4: общий вывод.

В ходе исследования нами была проведена предобработка данных - обработаны пропущенные значения, удалены явные и неявные дубликаты, изменены типы данных на необходимый, выделены категории данных для анализа.

Анализ показал, что вероятность возникновения долгов и просрочек зависит от ряда факторов: количества детей, семейного положения, уровня дохода и цели кредита.

1. **Количество детей**: Люди с большим количеством детей, особенно с четырьмя, имеют более высокую вероятность возникновения долгов, что может быть связано с большей финансовой нагрузкой. Наименьший риск наблюдается у людей без детей.
Люди с 4 детьми имеют самую высокую долю должников — 9.76%, что может свидетельствовать о значительной финансовой нагрузке.
У людей с 2 детьми также высокая доля должников — 9.45%.
Люди с 1 ребенком имеют долю должников 9.23%.
С 3 детьми — 8.18% должников.
Самая низкая доля должников у людей без детей — 7.54%.
У людей с 5 детьми доля должников составляет 0%, что может быть связано с малым количеством наблюдений в этой группе.

2. **Семейное положение**: Одинокие люди и те, кто живет в гражданском браке, чаще сталкиваются с долгами, тогда как женатые, разведенные и вдовцы имеют меньший риск, вероятно, из-за стабильности и финансовой поддержки. Наибольший риск долгов у одиноких людей — 9.76%.
Люди в гражданском браке имеют долю должников 9.31%.
У женатых и замужних людей доля должников составляет 7.56%.
Разведенные имеют долю должников 7.06%.
Вдовы и вдовцы имеют самую низкую долю должников — 6.62%.

3. **Уровень дохода**: Зависимость между уровнем дохода и возвратом кредита в срок действительно существует, но она лучше видна в более крупных группах (B и C). Категории B и D, с доходами от 30,001 до 1,000,000, демонстрируют меньшую вероятность просрочки по кредитам. Для категорий A и E выводы нельзя считать надежными из-за малых размеров выборки и необходимо более тщательное оценивание рисков перед выдачей кредита.

4. **Цель кредита**: Кредиты на автомобиль и образование наиболее подвержены просрочкам, что может быть связано с высокими расходами и финансовой нагрузкой. Напротив, кредиты на свадьбу и недвижимость имеют наименьшую вероятность просрочки, что свидетельствует о более ответственном подходе к этим финансовым обязательствам. Кредиты на автомобиль имеют наибольшую долю просрочек — 9.35%.
Кредиты на образование также имеют высокую долю просрочек — 9.25%.
Кредиты на свадьбу реже просрочиваются — 7.91%.
Самая низкая доля просрочек у кредитов на недвижимость — 7.26%.

**Заключение**: Результаты показывают, что кредитный риск сильно зависит от сочетания личных и финансовых факторов. Для банков и кредиторов важно учитывать эти зависимости при оценке кредитоспособности клиентов, чтобы минимизировать риски невозврата.

Эти выводы могут быть использованы для разработки более точных моделей кредитного скоринга и принятия решений в кредитной политике банков.


**На основе проведенного анализа можно предложить следующие рекомендации для компании:**

1. **Персонализация кредитных условий**:
   - Семейное положение: Для одиноких людей и тех, кто живет в гражданском браке, стоит рассмотреть более строгие условия кредитования, так как они имеют более высокий риск возникновения долгов.
   - Количество детей: Особое внимание следует уделить заемщикам с 4 детьми, предлагая им либо более низкие кредитные лимиты, либо обязательные консультации по финансовому планированию.

2. **Усиление контроля за высокорисковыми категориями заемщиков**:
   - Уровень дохода: Заемщики из категорий A и E требуют более тщательного анализа платежеспособности перед одобрением кредита из-за их малого числа и возможной вероятности просрочек.
   - Цель кредита: Кредиты на автомобиль и образование требуют дополнительных мер по снижению рисков, таких как более высокие требования к первоначальному взносу или страхованию кредита.

3. **Оптимизация кредитных предложений**:
   - Категории с низким риском: Заемщикам из категорий дохода B и D, а также тем, кто берет кредиты на свадьбу или недвижимость, можно предложить более выгодные условия кредитования, чтобы стимулировать их активность.

4. **Разработка улучшенных моделей кредитного скоринга**:
   - Интегрировать выявленные зависимости (количество детей, семейное положение, уровень дохода и цель кредита) в алгоритмы оценки рисков для более точного прогнозирования вероятности возврата кредита.

5. **Повышение финансовой грамотности**:
   - Ввести обязательные образовательные программы для высокорисковых групп (например, с низким доходом или большим количеством детей) по управлению личными финансами и планированию бюджета.

Эти рекомендации помогут компании снизить риски невозврата кредитов и улучшить качество портфеля клиентов, ориентируясь на выявленные зависимости и особенности различных групп заемщиков.


